### DISCUSSION TITLE
Can we Implement Stream Encryption and Decryption using S3.getObject().createReadStream?

### DISCUSSION THREAD

**Discussion by hanoj-budime** [openpgpjs/openpgpjs#1675] [Q&A]
OpenPGP.js version: 5.10.1 ( latest )
Affected platform (Browser or Node.js version): 14 +

https://github.com/openpgpjs/openpgpjs#streaming-encrypt-and-decrypt-string-data-with-pgp-keys

Streaming encrypt and decrypt Using AWS S3 stream object with PGP keys
`AWS.S3().getObject().createReadStream();` 

But in openPGP@latest version iam not able do it with this way ( S3.getObject().createReadStream() );

``` js
const objectStream= s3.getObject({ Bucket, Key: `test` }).createReadStream();
const encrypt = async (objectStream, encryptionKey) =>
  await openpgp.encrypt({
    message: await openpgp.createMessage({ binary: objectStream }),
    encryptionKeys: encryptionKey,
    format: "binary",
  });
```

``` js
const encryptedObjectStream= s3.getObject({ Bucket, Key: EncryptedDoc }).createReadStream();
const decryptedResponse = async (encryptedObjectStream, decryptionKey) =>
  await openpgp.decrypt({
    message: await openpgp.readMessage({ binaryMessage : encryptedObjectStream }),
    decryptionKeys: decryptionKey,
    format : "binary"
  });
```
- - - - 
### Error

```ps
UnhandledPromiseRejectionWarning: MalformedXML: The XML you provided was not well-formed or did not validate against our published schema
```

https://github.com/openpgpjs/openpgpjs/discussions/1472

**Comment by larabr** [comment]
The error is from the s3 API and the object it expects to store. Not really sure how we can help you here, without more context and/or code?

**Comment by paul-uz** [comment]
Did you ever figure this out?
